### PR TITLE
Refactor some common code out of salesforcedx-apex-debugger

### PR DESCRIPTION
**Pull request by JimSuplizio**
### What does this PR do?
Refactor some common code out of salesforcedx-apex-debugger into salesforcedx-utils-vscode.

Refactor the BaseCommand, RequestService, ForceConfigGet and ForceOrgDisplay 

The RequestSerivce is basically a REST command executor and is used by the Apex debugger and the Apex Replay Debugger. It was odd for the replay debugger to have a dependency on the Apex debugger package. I moved the RequestService and BaseCommands into the utils-vscode extension. Further, the ForceConfigGet and ForceOrgDisplay being CLI commands were also moved. Both were moved but only the ForceOrgDisplay is currently being used by multiple extensions.

The vast majority of files in this PR are changes to imports. 

The project.json file for salesforcedx-utils-vscode did require dependency updates for sinon, mock-spawn and request-light. Because of these changes the salesforcedx-vscode-replay-debugger/tsconfig.json needed to have "dom" added to it's list of libs.

### What issues does this PR fix or reference?
@W-4897179@

### CODE REVIEW

**Comment by vazexqi** [comment]
@JimSuplizio I noticed that you did 

  private _instanceUrl: string;
  private _accessToken: string;
  private _proxyUrl: string;
  private _proxyStrictSSL: boolean;
  private _proxyAuthorization: string;
  private _connectionTimeoutMs: number;

Please don't use that convention. Remove all the _ in front of field names. I didn't see this in an earlier commit.

File: packages/salesforcedx-vscode-replay-debugger/tsconfig.json [hunk 1]
```diff
@@ -2,9 +2,7 @@
   "compilerOptions": {
```
Comment (vazexqi) [inline packages/salesforcedx-vscode-replay-debugger/tsconfig.json]: I think a wrong formatter could have been used here. Could you make sure that is formatted via prettier?

File: packages/salesforcedx-utils-vscode/package.json [hunk 2]
```diff
@@ -17,16 +17,20 @@
     "@types/mocha": "^2.2.38",
```
Comment (vazexqi) [inline packages/salesforcedx-utils-vscode/package.json]: You added to salesforcedx-utils-vscode. Do you need to also remove from the other package.json?

**Comment by vazexqi** [comment]
@JimSuplizio - Could you grab the artifacts at https://circleci.com/gh/forcedotcom/salesforcedx-vscode/582#artifacts/containers/0 and ensure that when you run them you don't have any issues?

Even if the tests pass, the generated .vsix might not be packaged properly due to the way `npm prune` works. I'm still trying to figure out why but I ran into it in #386 while I was testing manually with the .vsix.

File: packages/salesforcedx-utils-vscode/src/constants.ts [hunk 3]
```diff
@@ -5,4 +5,6 @@
  * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
 
+export const CLIENT_ID = 'sfdx-vscode';
+export const DEFAULT_CONNECTION_TIMEOUT_MS = 20000;
```
Comment (vazexqi) [inline packages/salesforcedx-utils-vscode/src/constants.ts]: You moved this here but I don't see a removal. Where was this previously?

**Comment by vazexqi** [comment]
Yes apologies about https://github.com/forcedotcom/salesforcedx-vscode/pull/389#issuecomment-382855809

I see it was added by @DatGuyJonathan as part of https://github.com/forcedotcom/salesforcedx-vscode/commit/928f64bb22e82fcdbe7d35d8db55f6cdd8fab866

File: packages/salesforcedx-utils-vscode/package.json (hunk 2 above)
Comment (JimSuplizio) [reply packages/salesforcedx-utils-vscode/package.json]: I only moved a couple of classes and their respective tests. There are still a ton of tests (test, test:unit and test:integration) in salesforcedx-apex-debugger that require it.

File: packages/salesforcedx-utils-vscode/src/constants.ts (hunk 3 above)
Comment (JimSuplizio) [reply packages/salesforcedx-utils-vscode/src/constants.ts]: Removed. It'll be updated with this next push. I'm just waiting on comments from Jonathan so I don't needlessly trigger appveyor runs.

Comment (JimSuplizio) [reply packages/salesforcedx-utils-vscode/src/constants.ts]: Pushing the change up with that in it now.

**Comment by JimSuplizio** [comment]
Changes for @w-4897179@

**Pull request merged**